dragonfly/implementations/roach_daq_run_interface.py

```diff
@@ -183,19 +185,19 @@ def _start_data_taking(self, directory, filename):
         logger.info('Going to tell psyllid to start the run')
         payload = {'channel':self.channel_id, 'filename': os.path.join(directory, psyllid_filename), 'duration':duration}
         try:
-            self.provider.cmd(self.psyllid_interface, 'start_run', payload=payload)
-        except core.exceptions.DriplineError:
-            logger.error('Error from psyllid provider or psyllid')
+            self.provider.cmd(self.psyllid_interface, 'start_run', payload = payload)
+        except core.exceptions.DriplineError as e:
+            logger.critical('Error from psyllid provider or psyllid. Starting psyllid run failed')
             payload = {'channel': self.channel_id}
             try:
                 self.provider.cmd(self.daq_target, 'unblock_channel', payload = payload)
```

Comment:
Are there any cases when a channel will already be blocked when entering this `_start_data_taking` method?  Because you don't check if the channel is blocked before blocking, the `daq_target.block_channel` could do nothing (makes it True, when it was already True), and then unblocking may be problematic?  I don't know what checks have already been made before calling this, maybe this can't happen.

Specific use case: What if the `psyllid_interface.start_run` fails because a run is already going, then you wouldn't want to unblock the channel because there is still a legit run going. (I don't know if that can happen though).

Reply:
start_run calls _do_checks before calling _start_data_taking.
_do_checks already checks whether the channel is blocked (and raises an error if it is) so _start_data_taking doesn't need to check again.
_do_checks also checks whether there is a run ongoing.